R/learner_aorsf_surv_aorsf.R

```diff
@@ -0,0 +1,278 @@
+                                     lower = 1L,
+                                     tags = 'train'),
+
+                split_min_events = p_int(default = 5L,
+                                         lower = 1L,
+                                         tags = 'train'),
+
+                split_min_obs = p_int(default = 10,
+                                      lower = 1L,
+                                      tags = 'train'),
+
+                split_min_stat = p_dbl(default = 3.841459,
+                                       lower = 0,
+                                       tags = 'train'),
+
+                oobag_pred_type = p_fct(levels = c("none",
+                                                   "surv",
+                                                   "risk",
+                                                   "chf"),
+                                        default = "surv",
+                                        tags = 'train'),
+
+                importance = p_fct(levels = c("none",
+                                              "anova",
+                                              "negate",
+                                              "permute"),
+                                   default = "anova",
+                                   tags = 'train'),
+
+                oobag_pred_horizon = p_uty(tags = "train"),
+
+                oobag_eval_every = p_int(lower = 1,
+                                         tags = "train"),
+
+                attach_data = p_lgl(default = TRUE, tags = "train")
+
+              )
+
+              super$initialize(
+
+                id = "surv.aorsf",
+
+                packages = c("mlr3extralearners", "aorsf", "pracma"),
+
+                feature_types = c("integer", "numeric", "factor", "ordered"),
+
+                predict_types = c("crank", "distr"),
+
+                param_set = ps,
+
+                properties = c("oob_error", "importance"),
+
+                man = "mlr3extralearners::mlr_learners_surv.aorsf",
+
+                label = "Oblique Random Forest"
+
+              )
+            },
+
+            #' @description
+            #' OOB concordance error extracted from the model slot `oob_error`.
+            #' @return `numeric()`.
+            oob_error = function() {
+
+              nrows <- nrow(self$model$eval_oobag$stat_values)
+              1 - self$model$eval_oobag$stat_values[nrows, 1L]
+
+            },
+
+            #' @description
+            #' The importance scores are extracted from the model.
+            #' @return Named `numeric()`.
+            importance = function() {
+
+              if (is.null(self$model)) {
+                stopf("No model stored")
+              }
+
+              aorsf::orsf_vi(self$model, group_factors = TRUE)
+
+            }
+
+          ),
+
+          private = list(
+            .train = function(task) {
+
+              # start w/defaults
+              ps <- self$param_set$default
```

Comment:
Noted! I've initialized following the example linked and added a helper function to `.train` so that I can organize the input for `aorsf::fit()` and get the correct type of `control` object. In retrospect making the `control` input a function was not a good choice for compatibility with `mlr3`, whoops =]